widgetssdk/src/main/java/com/glia/widgets/core/dialog/DialogManager.java

```diff
@@ -10,15 +11,22 @@
 public final class DialogManager {
     @NonNull
     private final Callback callback;
-    private final Deque<DialogState> stateDeque;
+    @VisibleForTesting
+    final Deque<DialogState> stateDeque;
     private final DialogState none = new DialogState(Dialog.MODE_NONE);
 
     public DialogManager(@NonNull Callback callback) {
         this.callback = callback;
         stateDeque = new LinkedBlockingDeque<>();
     }
 
-    public void offer(DialogState state) {
+    /**
+     * Used to add state to the queue.
+     *
+     * It will emit a new state immediately. The previous state saved in the queue
+     * and will emit after the {@link #dismissCurrent()} called.
```

Comment:
Minor suggestion:
The previous is state saved in the queue and will be emitted after the {@link #dismissCurrent()} is called.